util/pace/util/initialization/allocator.py

```diff
@@ -117,3 +119,40 @@ def _allocate(
         except TypeError:
             data = allocator(shape, dtype=dtype)
         return Quantity(data, dims=dims, units=units, origin=origin, extent=extent)
+
+    def get_quantity_halo_spec(
```

Comment:
should we cache this?

Reply:
Maybe, but it's only called on init, and would need a bit of extra logic because you can't LRU cache on a list (not hashable).